pkg/object/redis.go

```diff
@@ -94,19 +95,45 @@ func (t *redisStore) ListAll(prefix, marker string) (<-chan Object, error) {
 		}
 		cursor = c
 	}
+
 	sort.Strings(keyList)
-	now := time.Now()
-	for _, key := range keyList {
-		data, err := t.rdb.Get(c, key).Bytes()
+
+	lKeyList := len(keyList)
+	for start := 0; start < lKeyList; start += batch {
+		end := start + batch
+		if end > lKeyList {
+			end = lKeyList
+		}
+
+		p := t.rdb.Pipeline()
+		for _, key := range keyList[start:end] {
+			p.StrLen(c, key)
+		}
+		cmds, err := p.Exec(ctx)
 		if err != nil {
-			if err == redis.Nil {
-				continue
-			}
 			return nil, err
 		}
 
-		// FIXME: mtime
-		objs <- &obj{key, int64(len(data)), now, strings.HasSuffix(key, "/")}
+		now := time.Now()
+		for idx, cmd := range cmds {
+			if intCmd, ok := cmd.(*redis.IntCmd); ok {
+				size, err := intCmd.Result()
+				if err != nil {
```

Comment:
Will err be Nil?

Reply:
err will not be nil but size will be 0 when key does not exist